code/modules/paperwork/paper.dm

```diff
@@ -1,160 +1,189 @@
 // large amount of fields creates a heavy load on the server, see updateinfolinks() and addtofield()
 #define MAX_FIELDS 50
 
+///Precached overlays commonly used by papers
+var/global/list/cached_paper_overlays
+
+/**Returns the cached paper overlay/underlay with the specified name. Also handles caching the images if needed.*/
+/proc/get_paper_overlay(var/name)
+	if(!LAZYLEN(global.cached_paper_overlays))
+		cache_paper_overlays()
+	return global.cached_paper_overlays[name]
+
+/**Caches the paper overlays/underlays we use frequently.*/
+/proc/cache_paper_overlays()
+	if(LAZYLEN(cached_paper_overlays))
+		return
+	LAZYSET(cached_paper_overlays, "paper_words",       overlay_image('icons/obj/bureaucracy.dmi', icon_state = "paper_words"))
+	LAZYSET(cached_paper_overlays, "paper_stack_words", overlay_image('icons/obj/bureaucracy.dmi', icon_state = "paper_stack_words"))
+	LAZYSET(cached_paper_overlays, "paper_talisman",    overlay_image('icons/obj/bureaucracy.dmi', icon_state = "paper_talisman", flags = RESET_COLOR))
+	LAZYSET(cached_paper_overlays, "sticky_words",      overlay_image('icons/obj/stickynotes.dmi', icon_state = "paper_words"))
+	LAZYSET(cached_paper_overlays, "photo_underlay",    overlay_image('icons/obj/bureaucracy.dmi', icon_state = "photo"))
```

Comment:
What's the point of this? SSoverlays probably will cache their appearances already, and even if not these images are not expensive.

Reply:
Not entirely sure about this? Its been a long time. I know that at the time I had been told to cache things on another PR. Not sure what's the maintainer team's consensus on caching?
But yeah, using add_overlay with the state name should be sufficient for a lot of those.

Reply:
Check with Loaf, if he prefers this then leave it. I would personally never do this though.

Reply:
kk